### PR TITLE
Affiliate system

### PR DESCRIPTION
**Pull request by nuevoalex**
Requirements described here:

https://docs.google.com/document/d/1kU2WNy-xwrcwQW_8Gu8VaNFy98mebKuQxPzKXmlemo4/edit#heading=h.z9lxxtgf2aqr

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). 💯
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/p37hw04**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

File: packages/augur-test/src/tests/state/getter/Users.test.ts [hunk 1]
```diff
@@ -358,7 +359,7 @@ describe('State API :: Users :: ', () => {
     await john.augur.contracts.shareToken.claimTradingProceeds(
       johnYesNoMarket.address,
       john.account.publicKey,
-      '0x0000000000000000000000000000000000000000'
+      formatBytes32String("")
```
Comment (robert-e-davidson3) [inline packages/augur-test/src/tests/state/getter/Users.test.ts:362]: This doesn't need to be `formatBytes32String("0x")`?

File: packages/augur-sdk/src/api/Market.ts [hunk 2]
```diff
@@ -68,10 +77,13 @@ export class Market {
 
   async createScalarMarket(params: CreateScalarMarketParams): Promise<ContractInterfaces.Market> {
     const universe = this.augur.contracts.universe;
+    // TODO
+    const affiliateValidator = params.affiliateValidator || "0x0000000000000000000000000000000000000000";
```
Comment (robert-e-davidson3) [inline packages/augur-sdk/src/api/Market.ts:81]: Use `NULL_ADDRESS`?

**Review by robert-e-davidson3 — APPROVED** [review]
Mostly over my head but what what I undestood lgtm.

**Pull request merged**